nestle.py

```diff
@@ -655,19 +732,23 @@ def set_options(self, options):
         self.enlarge = options.get('enlarge', 1.2)
 
     def update(self, pointvol):
+        self.empty_queue()
         self.ells = bounding_ellipsoids(self.points, pointvol=pointvol)
         for ell in self.ells:
             ell.scale_to_vol(ell.vol * self.enlarge)
+        self.fill_queue()
 
-    def new_point(self, loglstar):
+    def propose_point(self):
+        while True:
+            u = sample_ellipsoids(self.ells, rstate=self.rstate)
+            if np.all(u > 0.) and np.all (u < 1.):
+                break
+        return u
+
+    def new_point(self, loglstar, should_update):
         ncall = 0
         while True:
-            while True:
-                u = sample_ellipsoids(self.ells, rstate=self.rstate)
-                if np.all(u > 0.) and np.all(u < 1.):
-                    break
-            v = self.prior_transform(u)
-            logl = self.loglikelihood(v)
+            u, v, logl = self.get_point_value(should_update)
```

Comment:
This calls get_point_value a potentially arbitrary number of times, but if should_update is set get_point_value will fail after the queue empties. If you want to make should_update work, you need to handle two possibilities: (a) there is an eligible point (logl>=loglstar) in the queue, or (b) there is no eligible point remaining. In fact you always need to handle (b), since sooner or later you'll run out of eligible points. In this case new_point needs to return some kind of special value and its caller needs to do something to take into account that no eligible point was found.